### PR TITLE
Next Demo: React UI library integration (part 1)

### PR DESCRIPTION
**Pull request by jmderby**
## Description

This PR depends on `jonathan/client-sdk-react-ui-auth-provider` and contains the changes to use the new hook and auth provider.

This PR also contains various UI tweaks to correspond to the mock design better, thanks to the help of Pablo. 

## Test plan

Regular demo flow still works as expected.

### CODE REVIEW

File: packages/client/auth-core/package.json [hunk 1]
```diff
@@ -34,11 +28,12 @@
         "LICENSE"
     ],
     "scripts": {
-        "build": "tsup src/clientIndex.ts src/serverIndex.ts --clean --external react,react-dom --format esm,cjs --outDir ./dist --minify --dts --sourcemap",
-        "dev": "tsup src/clientIndex.ts src/serverIndex.ts --clean --external react,react-dom --format esm,cjs --outDir ./dist --dts --sourcemap --watch",
+        "build": "tsup src/index.ts --clean --format esm,cjs --outDir ./dist --minify --dts --sourcemap",
+        "dev": "tsup src/index.ts --clean --format esm,cjs --outDir ./dist --dts --sourcemap --watch",
```
Comment (jmderby) [inline packages/client/auth-core/package.json:32]: not sure if we need file changes :thinking:

File: packages/client/auth-core/src/contexts/authlogin.tsx [hunk 2]
```diff
@@ -17,27 +17,29 @@ const AuthContext = createContext<AuthContextType>({
...
     children: ReactNode;
+    embeddedWallets: {
+        type: "evm-smart-wallet";
+        defaultChain: "polygon-amoy" | "base-sepolia";
+        createOnLogin: "all-users" | "off";
+    };
+};
+
+const getJwtFromCookie = (): string | null => {
+    if (typeof document === "undefined") return null; // Check if we're on the client-side
```
Comment (jmderby) [inline packages/client/auth-core/src/contexts/authlogin.tsx:32]: todo: reminder to check in with this

File: apps/wallets/smart-wallet/next/src/app/globals.css [hunk 3]
```diff
@@ -2,8 +2,6 @@
 @tailwind components;
 @tailwind utilities;
 
-@import url("https://fonts.googleapis.com/css2?family=Poppins:wght@400;500;700&display=swap");
-
```
Comment (jmderby) [inline apps/wallets/smart-wallet/next/src/app/globals.css:6]: replaced Poppins theme with Inter

File: apps/wallets/smart-wallet/next/src/app/page.tsx [hunk 4]
```diff
@@ -1,27 +1,24 @@
...
+import { useState } from "react";
 
-import { useAuth } from "../hooks/useAuth";
+import { useAuth } from "@crossmint/client-sdk-react-ui";
 
 export default function Home() {
-    const { isLoading, authedUser } = useAuth();
-    const { toasts } = useToast();
+    const { jwt, wallet } = useAuth();
+    const [nftSuccessfullyMinted, setNftSuccessfullyMinted] = useState(false);
```
Comment (jmderby) [inline apps/wallets/smart-wallet/next/src/app/page.tsx:15]: added explicit state to determine successful mint

File: packages/client/auth-core/package.json [hunk 5]
```diff
@@ -7,14 +7,8 @@
     "sideEffects": false,
```
Comment (jmderby) [inline packages/client/auth-core/package.json:7]: Not sure if we require changes in this file.

File: apps/wallets/smart-wallet/next/package.json [hunk 6]
```diff
@@ -9,6 +9,8 @@
         "start": "next start"
     },
     "dependencies": {
+        "@crossmint/client-sdk-auth-core": "workspace:*",
```
Comment (devlyn37) [inline apps/wallets/smart-wallet/next/package.json:12]: We should be able to delete this one now

File: apps/wallets/smart-wallet/next/src/app/template.tsx [hunk 7]
```diff
@@ -1,17 +1,44 @@
...
+    // causes a hydration error when the auth modal is opened for whatever reason.
+    // Checking for mount here also prevents the auth provider from throwing a "Smart Wallet SDK should only be used client side." error.
+    return !isMounted ? (
+        <>
             <Header />
             <Toaster />
             {children}
-        </div>
+        </>
+    ) : (
```
Comment (devlyn37) [inline apps/wallets/smart-wallet/next/src/app/template.tsx:31]: Seems weird that this isn't already the default behavior XD

Comment (devlyn37) [reply apps/wallets/smart-wallet/next/src/app/template.tsx:31]: Is this is the case when you do a production build too?

Comment (devlyn37) [reply apps/wallets/smart-wallet/next/src/app/template.tsx:31]: next is sometimes weird in dev mode

File: apps/wallets/smart-wallet/next/src/app/wallet/page.tsx [hunk 8]
```diff
@@ -63,28 +61,24 @@ const SkeletonLoader = () => {
...
+    const { wallet, isLoadingWallet } = useAuth();
 
     const { data, isLoading } = useQuery({
         queryKey: ["smart-wallet"],
         queryFn: async () => {
-            const authedJWT = await getAuthedJWT();
-            const wallet = await createOrGetPasskeyWallet(authedJWT as unknown as string);
-            const nfts = await wallet?.nfts();
-            return (nfts || []) as NFT[];
+            if (wallet && !isLoadingWallet) {
```
Comment (devlyn37) [inline apps/wallets/smart-wallet/next/src/app/wallet/page.tsx:69]: do we need the `isLoadingWallet` check here?

File: apps/wallets/smart-wallet/next/README.md
Comment (devlyn37) [inline apps/wallets/smart-wallet/next/README.md:1]: v/ nice

File: apps/wallets/smart-wallet/next/src/components/header.tsx [hunk 9]
```diff
@@ -1,66 +1,83 @@
...
+                                              wallet.address.substring(wallet.address.length - 3, wallet.address.length)
+                                            : "Loading..."}
+                                    </Typography>
+                                </div>
+                                <Avatar className="h-9 w-9">
+                                    <AvatarImage alt="User Avatar" src="" />
+                                    <AvatarFallback className="bg-skeleton">
+                                        <User className="h-5 w-5" />
+                                    </AvatarFallback>
+                                </Avatar>
```
Comment (devlyn37) [inline apps/wallets/smart-wallet/next/src/components/header.tsx:59]: double checking this is the same right color (same as the background)?

File: packages/client/auth-core/package.json
Comment (devlyn37) [inline packages/client/auth-core/package.json:1]: I don't think so, we should revert

File: packages/client/auth-core/src/contexts/authlogin.tsx (hunk 2 above)
Comment (devlyn37) [reply packages/client/auth-core/src/contexts/authlogin.tsx:32]: I think it's needed, but should be opened in a separate PR imo

File: packages/client/auth-core/src/utils/constants.ts
Comment (devlyn37) [inline packages/client/auth-core/src/utils/constants.ts:1]: need to revert this change

File: packages/client/wallets/smart-wallet/src/blockchain/chains.ts
Comment (devlyn37) [inline packages/client/wallets/smart-wallet/src/blockchain/chains.ts:1]: should revert this change

File: packages/client/auth-core/src/index.ts
Comment (devlyn37) [inline packages/client/auth-core/src/index.ts:1]: Don't think this is still needed

File: apps/wallets/smart-wallet/next/src/components/header.tsx (hunk 9 above)
Comment (jmderby) [reply apps/wallets/smart-wallet/next/src/components/header.tsx:59]: it's the same color as the wallet button background

Comment (jmderby) [reply apps/wallets/smart-wallet/next/src/components/header.tsx:59]: Screenshot from 2024-08-14 14-56-54

File: packages/client/auth-core/package.json
Comment (jmderby) [reply packages/client/auth-core/package.json:1]: Changes will be in the part 2 PR

File: apps/wallets/smart-wallet/next/src/app/template.tsx (hunk 7 above)
Comment (jmderby) [reply apps/wallets/smart-wallet/next/src/app/template.tsx:31]: Nextjs is definitely weird in dev mode. iirc building would throw because of that error.

File: apps/wallets/smart-wallet/next/src/components/header.tsx (hunk 9 above)
Comment (devlyn37) [reply apps/wallets/smart-wallet/next/src/components/header.tsx:60]: v/ nice

**Review by devlyn37 — APPROVED** [review]
Nice!
